### PR TITLE
Removing the default sampling frequency from create_from_X_y, #143.

### PR DESCRIPTION
**Pull request by vytjan**
Fixes the #143 
Also, not sure if it's fine to swap the order of arguments of the create_from_X_y, but as drop_last_window seems to be an optional, it could be better to assign a default value for it maybe?

And finally, since 4 of us (@Ann-KathrinKiessner, @dcwil, @HenrikBons) worked on this small issue for introduction, we'd like to know if (or how) we should add ourselves to the author's list of the respective files.

### CODE REVIEW

**Comment by robintibor** [comment]
uh why did you swap argument order? `drop_last_window` should not have a default window, we have discussed about this before. Different users may have different expectations there and it is good if they have to think about this argument once.

**Comment by robintibor** [comment]
you can swap the argument order back and just change the function calls accordingly

**Comment by vytjan** [comment]
Thanks for the feedback, just swapped the order back.

File: braindecode/datasets/xy.py [hunk 1]
```diff
@@ -26,12 +26,13 @@ def create_from_X_y(
         list of pre-cut trials as n_trials x n_channels x n_times
     y: array-like
         targets corresponding to the trials
-    sfreq: common sampling frequency of all trials
-    ch_names: array-like
-        channel names of the trials
+    sfreq: int
```
Comment (agramfort) [inline braindecode/datasets/xy.py:29]:
```suggestion
    sfreq: float
```

File: braindecode/datasets/xy.py [hunk 2]
```diff
@@ -26,12 +26,13 @@ def create_from_X_y(
...
+        common sampling frequency of all trials
```
Comment (agramfort) [inline braindecode/datasets/xy.py:30]:
```suggestion
        Sampling frequency of signals.
```

File: braindecode/datasets/xy.py [hunk 3]
```diff
@@ -26,12 +26,13 @@ def create_from_X_y(
...
     drop_last_window: bool
         whether or not have a last overlapping window, when
         windows/windows do not equally divide the continuous signal
+    ch_names: array-like
+        channel names of the trials
```
Comment (agramfort) [inline braindecode/datasets/xy.py:35]:
```suggestion
        Names of the channels.
```

File: test/acceptance_tests/test_cropped_decoding.py [hunk 4]
```diff
@@ -88,12 +88,12 @@ def test_cropped_decoding():
     # Perform forward pass to determine how many outputs per input
     n_preds_per_input = get_output_shape(model, in_chans, input_window_samples)[2]
 
-    train_set = create_from_X_y(X[:60], y[:60],
+    train_set = create_from_X_y(X[:60], y[:60], 100,
```
Comment (agramfort) [inline test/acceptance_tests/test_cropped_decoding.py:91]:
```suggestion
    train_set = create_from_X_y(X[:60], y[:60], sfreq=100,
```

File: test/acceptance_tests/test_cropped_decoding.py [hunk 5]
```diff
@@ -88,12 +88,12 @@ def test_cropped_decoding():
...
                                 drop_last_window=False,
                                 window_size_samples=input_window_samples,
                                 window_stride_samples=n_preds_per_input)
 
-    valid_set = create_from_X_y(X[60:], y[60:],
+    valid_set = create_from_X_y(X[60:], y[60:], 100,
```
Comment (agramfort) [inline test/acceptance_tests/test_cropped_decoding.py:96]:
```suggestion
    valid_set = create_from_X_y(X[60:], y[60:], sfreq=100,
```

File: test/acceptance_tests/test_eeg_classifier.py [hunk 6]
```diff
@@ -148,12 +148,12 @@ def test_eeg_classifier():
     out = model(test_input)
     n_preds_per_input = out.cpu().data.numpy().shape[2]
 
-    train_set = create_from_X_y(X[:48], y[:48],
+    train_set = create_from_X_y(X[:48], y[:48], 100,
```
Comment (agramfort) [inline test/acceptance_tests/test_eeg_classifier.py:151]:
```suggestion
    train_set = create_from_X_y(X[:48], y[:48], sfreq=100,
```

File: test/acceptance_tests/test_eeg_classifier.py [hunk 7]
```diff
@@ -148,12 +148,12 @@ def test_eeg_classifier():
...
                                 drop_last_window=False,
                                 window_size_samples=input_window_samples,
                                 window_stride_samples=n_preds_per_input)
 
-    valid_set = create_from_X_y(X[48:60], y[48:60],
+    valid_set = create_from_X_y(X[48:60], y[48:60], 100,
```
Comment (agramfort) [inline test/acceptance_tests/test_eeg_classifier.py:156]:
```suggestion
    valid_set = create_from_X_y(X[48:60], y[48:60], sfreq=100,
```

File: test/unit_tests/training/test_scoring.py [hunk 8]
```diff
@@ -87,7 +87,7 @@ def test_cropped_trial_epoch_scoring():
         predictions_cases, y_true_cases, expected_accuracies_cases
     ):
         dataset_valid = create_from_X_y(
-            np.zeros((4, 1, 10)), np.concatenate(y_true),
+            np.zeros((4, 1, 10)), np.concatenate(y_true), 100,
```
Comment (agramfort) [inline test/unit_tests/training/test_scoring.py:90]:
```suggestion
            np.zeros((4, 1, 10)), np.concatenate(y_true), sfreq=100,
```

File: braindecode/datasets/xy.py [hunk 9]
```diff
@@ -14,7 +14,7 @@
 
 
 def create_from_X_y(
-        X, y, drop_last_window, sfreq=None, ch_names=None, window_size_samples=None,
+        X, y, sfreq, drop_last_window, ch_names=None, window_size_samples=None,
```
Comment (agramfort) [inline braindecode/datasets/xy.py:17]: this is an API change.

@robintibor deprecation cycle or you can live with it?

**Comment by vytjan** [comment]
@agramfort thank you for the corrections. 
Apparently, before I didn't manage to amend my changes to the commit (basically all your corrections, which @Ann-KathrinKiessner, @dcwil, @HenrikBons and I have agreed on).

**Comment by robintibor** [comment]
Can you still add to what's new page? https://github.com/vytjan/braindecode/blob/mandatory_sfreq/docs/whats_new.rst create new subheading of current version API changes (copy heading from below), and add a sentence about this, also add all your names

**Comment by vytjan** [comment]
> Can you still add to what's new page? https://github.com/vytjan/braindecode/blob/mandatory_sfreq/docs/whats_new.rst create new subheading of current version API changes (copy heading from below), and add a sentence about this, also add all your names

Added the PR info to the what's new section (under the Current (0.5.2.dev0) tag, correct?).

**Review by agramfort — APPROVED** [review]

**Comment by robintibor** [comment]
Looks good! Merged! Thanks so much! @vytjan @Ann-KathrinKiessner @dcwil @HenrikBons

**Pull request merged**